src/libopensc/card-cac.c

```diff
@@ -606,13 +604,19 @@ static int cac_read_binary(sc_card_t *card, unsigned int idx,
 		/* SPICE smart card emulator only presents CAC-1 cards with the old CAC-1 interface as
 		 * certs. If we are a cac 1 card, use the old interface */
 		r = cac_cac1_get_certificate(card, &val, &val_len);
+		if (r == SC_ERROR_INS_NOT_SUPPORTED) {
+			/* The CACv1 instruction is not recognized. Try with CACv2 */
+			card->type = SC_CARD_TYPE_CAC_II;
+			goto cac2;
```

Comment:
Please avoid `goto` for anything else than cleanup. See https://github.com/frankmorgner/OpenSC/commit/1fad8ea512346f2fb7f5b44619f4ff9a0b3ade5e for a suggestion.